### ISSUE TITLE
Explicitly Track Gas Ejected by Supernovae/Quasar

### ISSUE THREAD

**Issue by jacobseiler** [jacobseiler/self_consistent_SAGE#9]
Currently the `EjectedGas` reservoir is ambivalent to how the gas was ejected.  However it could be useful to explicitly track which processes ejected gas to determine the relative impact of these two processes.

**Comment by jacobseiler** [comment]
So there's an issue here regarding how to treat the gas when it's reincorporated back into the galaxy.  Whilst we can track through which channel gas was ejected, it's more uncertain about when each channel of gas becomes reincorporated.

My current running idea is to have it so the reincorporated gas is drawn equally from both channels.  Then, in the case that one channel does not have sufficient gas I simply just draw the remaining from the other.

**Comment by jacobseiler** [comment]
There was numerical issues regarding subtracting gas from the two channels.  Within `model_infall.c`, if the halo has lost gas over a time step, the baryons are first subtracted from the `EjectedMass` reservoir.  

However for galaxies with very low masses (the ejected reservoir had ~2e3Msun) the numerical precision would cause problems when I required that `EjectedMassSN` and `EjectedMassQSO` reservoirs to be `>=0.0`.  This was solved by putting a numerical tolerance in and saying `if the galaxy has low mass, just fudge the tracking`.
